tsdb/cursor.go

```diff
@@ -71,26 +59,32 @@ func (mc *multiCursor) Seek(seek []byte) (key, value []byte) {
 
 	heap.Init(&h)
 	mc.heap = h
-	mc.prev = nil
+	mc.prev = EOF
 
 	return mc.pop()
 }
 
-func (mc *multiCursor) Direction() Direction { return mc.direction }
+// Ascending returns the direction of the first cursor.
+func (mc *multiCursor) Ascending() bool {
+	if len(mc.cursors) == 0 {
+		return true
+	}
+	return mc.cursors[0].Ascending()
+}
 
 // Next returns the next key/value from the cursor.
-func (mc *multiCursor) Next() (key, value []byte) { return mc.pop() }
+func (mc *multiCursor) Next() (int64, interface{}) { return mc.pop() }
 
 // pop returns the next item from the heap.
 // Reads the next key/value from item's cursor and puts it back on the heap.
-func (mc *multiCursor) pop() (key, value []byte) {
+func (mc *multiCursor) pop() (key int64, value interface{}) {
 	// Read items until we have a key that doesn't match the previously read one.
 	// This is to perform deduplication when there's multiple items with the same key.
 	// The highest priority cursor will be read first and then remaining keys will be dropped.
 	for {
-		// Return nil if there are no more items left.
+		// Return EOF marker if there are no more items left.
```

Comment:
`nil` isn't allowed now that we're using `int64` so I wanted to declare the marker more explicitly than simply doing `-1`.